include/time_series/internal/base.hxx

```diff
@@ -1,6 +1,13 @@
 // Copyright (c) 2019 Max Planck Gesellschaft
 // Vincent Berenz
 
+// throws an exception if SIGINT is received
+#define TS_THROW_IF_SIGINT                                    \
```

Comment:
why a macro and not a "normal" method ?

Reply:
Good point.  I can't remember what I was thinking, a normal function should work just as well...  I'll change it.